### PR TITLE
OTLP gRPC log exporter

### LINKED ISSUE #1024: [Logs] Implement `OtlpGrpcLogExporter` and `OtlpHttpLogExporter`
We need OTLP Log exporter of C++ in our company. But it seems has not plan for it right now.
Can I create a PR to implement `OtlpGrpcLogExporter` and `OtlpHttpLogExporter` to export `opentelemetry.proto.collector.logs.v1.ExportLogsServiceRequest` 

> https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/collector/logs/v1/logs_service.proto

### PR DESCRIPTION
**Pull request by ThomsonTan**
Fixes #1024

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,18 @@ if(WITH_OTLP_GRPC)
...
+
+  add_library(opentelemetry_exporter_otlp_grpc_log
+              src/otlp_grpc_log_exporter.cc)
+
+  set_target_properties(opentelemetry_exporter_otlp_grpc_log
+                        PROPERTIES EXPORT_NAME otlp_grpc_log_exporter)
+
+  target_link_libraries(
+    opentelemetry_exporter_otlp_grpc_log
+    PUBLIC opentelemetry_otlp_recordable protobuf::libprotobuf gRPC::grpc++)
```
Comment (owent) [inline exporters/otlp/CMakeLists.txt:39]: `protobuf::libprotobuf` is already PUBLIC linked by `opentelemetry_proto` , and `opentelemetry_proto` is PUBLIC linked by `opentelemetry_otlp_recordable`. Maybe PUBLIC linking `protobuf::libprotobuf` into `opentelemetry_exporter_otlp_grpc_log` and `opentelemetry_exporter_otlp_grpc` is redundant?

Comment (ThomsonTan) [reply exporters/otlp/CMakeLists.txt]: Good point. I removed `protobuf::libprotobuf` for both `opentelemetry_exporter_otlp_grpc` and `opentelemetry_exporter_otlp_grpc_log` targets.

File: exporters/otlp/src/otlp_grpc_exporter.cc [hunk 2]
```diff
@@ -68,15 +67,25 @@ std::unique_ptr<proto::collector::trace::v1::TraceService::Stub> MakeServiceStub
...
   }
-  return proto::collector::trace::v1::TraceService::NewStub(channel);
+
+  return channel;
+}
+
+/**
+ * Create service stub to communicate with the OpenTelemetry Collector.
+ */
+std::unique_ptr<proto::collector::trace::v1::TraceService::Stub> MakeTraceServiceStub(
```
Comment (lalitb) [inline exporters/otlp/src/otlp_grpc_exporter.cc:77]: Why to split this function ?

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_log_exporter.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+#  include "opentelemetry/exporters/otlp/protobuf_include_prefix.h"
+#  include "opentelemetry/proto/collector/logs/v1/logs_service.grpc.pb.h"
+#  include "opentelemetry/exporters/otlp/protobuf_include_suffix.h"
+
+// clang-format on
+
+#  include "opentelemetry/exporters/otlp/otlp_environment.h"
+#  include "opentelemetry/exporters/otlp/otlp_grpc_exporter_options.h"
+#  include "opentelemetry/exporters/otlp/otlp_recordable.h"
+#  include "opentelemetry/exporters/otlp/otlp_recordable_utils.h"
```
Comment (lalitb) [inline exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_log_exporter.h:18]: Do we need all these headers - at least `otlp_recordable.h` and `otlp_recordable_utils.h` shouldn't be part of public API.

File: exporters/otlp/src/otlp_grpc_log_exporter.cc [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      ssl_opts.pem_root_certs = get_file_contents((options.ssl_credentials_cacert_path).c_str());
+    }
+    channel = grpc::CreateChannel(grpc_target, grpc::SslCredentials(ssl_opts));
+  }
+  else
+  {
+    channel = grpc::CreateChannel(grpc_target, grpc::InsecureChannelCredentials());
+  }
+
+  return channel;
```
Comment (lalitb) [inline exporters/otlp/src/otlp_grpc_log_exporter.cc:90]: Ok. now I understand the split. Do you think we can move both these helper functions to common place as part of this PR if this is not a lot of effort?

**Review by lalitb — APPROVED** [review]

File: exporters/otlp/src/otlp_grpc_exporter.cc (hunk 2 above)
Comment (ThomsonTan) [reply exporters/otlp/src/otlp_grpc_exporter.cc:77]: Split of `MakeTraceServiceStub` was meant to move `MakeGrpcChannel` to common lib, but then it also adds gRPC dependence to the common lib. I suggest we improve this in a separate PR.

File: exporters/otlp/src/otlp_grpc_log_exporter.cc (hunk 4 above)
Comment (ThomsonTan) [reply exporters/otlp/src/otlp_grpc_log_exporter.cc:90]: Yes, move it to common was the intention, but it moves gRPC dependence there as well. How about we do that in a separate PR?

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_log_exporter.h (hunk 3 above)
Comment (ThomsonTan) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_log_exporter.h]: Good catch, removed both headers.

**Pull request merged**